sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -4082,18 +4082,21 @@ object ApplyCharTypePadding extends Rule[LogicalPlan] {
 
```

Comment:
shall we simply skip this rule if the conf is true? e.g. in the `apply` method, we return the input plan if  `SQLConf.get.charVarcharAsString` is true.

Reply:
+1